app/services/doc_auth_client/response.rb

```diff
@@ -0,0 +1,39 @@
+module DocAuthClient
+  class Response
+    attr_reader :errors, :exception, :extra, :pii_from_doc
+
+    def initialize(success:, errors: [], exception: nil, extra: {}, pii_from_doc: {})
+      @success = success
+      @errors = errors
+      @exception = exception
+      @extra = extra
+      @pii_from_doc = pii_from_doc
+    end
+
+    # rubocop:disable Metrics/AbcSize
+    def merge(other)
+      Response.new(
+        success: success? && other.success?,
+        errors: errors + other.errors,
```

Comment:
maybe do `[*errors, *other.errors]` in case one of them is nil?

Reply:
Ah, good call